src/android/patches/makeSettingsPatch.js

```diff
@@ -2,7 +2,7 @@ const path = require('path');
 
 module.exports = function makeSettingsPatch(name, dependencyConfig, projectConfig) {
   const relative = path.relative(
-    path.dirname(projectConfig.settingsGradlePath),
+    projectConfig.settingsGradlePath,
```

Comment:
As I already said, mocha tests are corrupted because you have changed the line "10" (settingsGradlePath pointing to a file) to line "8" where it points to a folder (which is wrong, because rnpm sends over a **path**, not **location**)